src/TropicalGeometry/curve.jl

```diff
@@ -42,26 +44,467 @@ end
 @doc Markdown.doc"""
     TropicalCurve{M, EMB}()
 
-Construct a tropical curve from a list of edges and a vector of their lengths.
-If the curve is embedded, vertices must be points in $\mathbb R^n$.
-If the curve is abstract, vertices must be 1, ..., n.
+Construct a tropical curve from a polyhedral complex. 
+If the curve is embedded, vertices must are points in $\mathbb R^n$.
+If the curve is abstract, the polyhedral complex is empty, vertices must be 1, ..., n, 
+and the graph is given as attribute. 
 
 # Examples
+```jldoctest
+julia> IM = IncidenceMatrix([[1,2],[1,3],[1,4]])
+3×4 IncidenceMatrix
+[1, 2]
+[1, 3]
+[1, 4]
+
+
+julia>  VR = [0 0; 1 0; -1 0; 0 1]
+4×2 Matrix{Int64}:
+  0  0
+  1  0
+ -1  0
+  0  1
+
+julia> PC = PolyhedralComplex{fmpq}(IM, VR)
+A polyhedral complex in ambient dimension 2
+
+julia> TC = TropicalCurve{min}(PC)
+A tropical curve in 2-dimensional Euclidean space
+
+julia> abs_TC = TropicalCurve{min}(IM)
+An abstract tropical curve
+```
+"""
+function TropicalCurve{M}(PC::PolyhedralComplex) where {M}
+   @assert dim(PC)==1 "The polyhedral complex is not of dimenion 1."
+   return TropicalCurve{M, true}(PC)
+end
+
+
+function TropicalCurve{M}(graph::IncidenceMatrix) where {M}
+    # Columns correspond to nodes
+    # Rows correpons to edges
+    empty = PolyhedralComplex(Polymake.fan.PolyhedralComplex())
+    result = TropicalCurve{M, false}(empty)
+    set_attribute!(result, :graph, graph)    
+    return result
+end
+
+@doc Markdown.doc"""
+    graph(tc::TropicalCurve{M, EMB})
+
+Returns the graph of an abstract tropical curve `tc`. 
```

Comment:
We prefer imperative form, following [the Julia conventions for this](https://docs.julialang.org/en/v1/manual/documentation/)
```suggestion
Return the graph of an abstract tropical curve `tc`. 
```